content/java/concepts/comparator/comparator.md

```diff
@@ -0,0 +1,125 @@
+---
+Title: 'Comparator'
+Description: 'The Comparator interface is used to order objects of an arbitrary class.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Collections'
+  - 'Comparison'
+  - 'Data Structures'
+  - 'Sort'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`Comparator`** interface is used to order objects of an arbitrary [class](https://www.codecademy.com/resources/docs/java/classes). It is not to be confused with the [`Comparable`](https://www.codecademy.com/resources/docs/java/comparable) interface, which is implemented by the class to be sorted. The `Comparator` interface is implemented in a separate class.
+
+## Syntax
+
+```pseudo
+class MyComparator implements Comparator<MyClass> {
+  @Override public int compare(MyClass a, MyClass b)
+  {
+    // Compare logic
+    ...
+    return result;
+  }
+}
+```
+
+Applying the `Comparator` interface to a class, `MyComparator`, requires the `implements` keyword (e.g., `Comparator<MyClass>`). This interface has a `.compare()` method that returns an `int` value based on whether two `MyClass` instances, `a` and `b`, can be logically sorted.
+
+| Return Value | Meaning                                        |
+| :----------: | ---------------------------------------------- |
+|    `>= 1`    | first object instance > second object instance |
+|     `0`      | first object instance = second object instance |
+|   `<= -1`    | first object instance < second object instance |
```

Comment:
```suggestion
|    >= 1    | first object instance > second object instance |
|     0      | first object instance = second object instance |
|   <= -1    | first object instance < second object instance |
```

Although  this is return output, I'm not sure if they should necessarily be in back-ticks (to [a point made in a separate PR](https://github.com/Codecademy/docs/pull/840#discussion_r919231406)).